packages/api/src/errors.js

```diff
@@ -0,0 +1,111 @@
+import { ErrorCode as MagicErrors } from '@magic-sdk/admin'
+import { JSONResponse } from './utils/json-response.js'
+
+export class HTTPError extends Error {
+  /**
+   *
+   * @param {string} message
+   * @param {number} [status]
+   */
+  constructor (message, status = 500) {
+    super(message)
+    this.name = 'HTTPError'
+    this.status = status
+  }
+
+  /**
+   * @param {string} message
+   * @param {number} [status]
+   * @returns {never}
+   */
+  static throw (message, status) {
+    throw new this(message, status)
+  }
```

Comment:
I think `throw new HTTPError('boom', 500)` is fine, and would argue that `HTTPError.throw('boom', 500)` is not any more convenient.

```suggestion
```